### DISCUSSION TITLE
Safely producing TOML

### DISCUSSION THREAD

**Discussion by ShellWowza** [hairyhenderson/gomplate#1194] [Q&A] [answered]
Hi,

I have an input YAML datasource and I'm trying to output a TOML file containing a transformed/subsetted version of that datasource - essentially a different format.

I'm trying to work out how to correctly escape my strings. The obvious thing to do would be to use data.ToTOML, but I can't find a way to create an object to pass into that. Any ideas here?

Thanks,
Shell

**Comment by argl** [comment]
I am also wondering how this works.

**Comment by hairyhenderson** [comment, marked as answer]
Hi @ShellWowza, thanks for posting this question.

There's an example that's related to what you're looking for in [the docs](https://docs.gomplate.ca/functions/data/#data-totoml) which might help.

Internally the [`ds`/`datasource` function](https://docs.gomplate.ca/functions/data/#datasource) will convert from YAML to an internal object representation, so you don't need to parse the YAML at all. Something like this should work:

```console
$ gomplate -d in=./inputfile.yaml -i '{{ ds "in" | data.ToTOML }}'
```

Why don't you try that out and see if it works for you - and please post back!

**Reply by hairyhenderson** [reply]
@ShellWowza I'm assuming that "no news is good news" so I'm marking this as the answer 😉
